beam/src/main/scala/magnolify/beam/BeamSchemaType.scala

```diff
@@ -0,0 +1,283 @@
+/*
+ * Copyright 2024 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package magnolify.beam
+
+import magnolia1.*
+import magnolify.shared.*
+import org.apache.beam.sdk.schemas.Schema
+import org.apache.beam.sdk.schemas.Schema.{Field, FieldType}
+import org.apache.beam.sdk.values.Row
+import com.google.protobuf.ByteString
+import magnolify.shims.FactoryCompat
+import org.apache.beam.sdk.schemas.logicaltypes
+
+import java.nio.ByteBuffer
+import java.util as ju
+import scala.collection.compat.*
+import scala.collection.concurrent
+import scala.jdk.CollectionConverters.*
+
+// https://beam.apache.org/documentation/programming-guide/#schema-definition
+sealed trait BeamSchemaType[T] extends Converter[T, Row, Row] {
```

Comment:
The naming dose not match with convention of other modules.

```suggestion
sealed trait RowType[T] extends Converter[T, Row, Row] {
```

Reply:
Done